cmd/kube-rbac-proxy/app/sanitazion.go

```diff
@@ -0,0 +1,70 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+)
+
+// SanitizingFilter implements the LogFilter interface from klog with custom functions to detect and mask tokens.
+type SanitizingFilter struct{}
+
+// Filter is the filter function for non-formatting logging functions of klog.
+func (sf *SanitizingFilter) Filter(args []interface{}) []interface{} {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				args[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return args
+}
+
+// FilterF is the filter function for formatting logging functions of klog.
+func (sf *SanitizingFilter) FilterF(format string, args []interface{}) (string, []interface{}) {
+	for i, v := range args {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				args[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return format, args
+}
+
+// FilterS is the filter function for structured logging functions of klog.
+func (sf *SanitizingFilter) FilterS(msg string, keysAndValues []interface{}) (string, []interface{}) {
+	for i, v := range keysAndValues {
+		if strValue, ok := v.(string); ok {
+			if containsTokenReview(strValue) {
+				keysAndValues[i] = maskTokenInLog(strValue)
+			}
+		}
+	}
+	return msg, keysAndValues
+}
+
+func containsTokenReview(logStr string) bool {
+	return bytes.Contains([]byte(logStr), []byte(`"kind":"TokenReview"`))
+}
+
+func maskTokenInLog(logStr string) string {
+	var logMap map[string]interface{}
+	if err := json.Unmarshal([]byte(logStr), &logMap); err != nil {
+		return logStr
+	}
+
+	if spec, ok := logMap["spec"].(map[string]interface{}); ok {
```

Comment:
This is pretty static. It will always try to go for `.spec.token` and try to mask it. So in order for this to be a valid assumption, we would need to go through the k8s.io API and verify that all tokens that can be delivered over the wire, are in this path... otherwise a regex might be better. The regex is safer, but the static path is most probably quicker.

Reply:
I’ve reviewed all the api-resources, and it turns out that only `TokenReview` contains a token field, so we can keep the current approach